src/Forms/NumericField.php

```diff
@@ -132,71 +205,105 @@ public function getSchemaValidation()
     }
 
     /**
-     * Extracts the number value from the localised string value.
+     * Get internal database value
      *
-     * @return string
+     * @return int|float
      */
     public function dataValue()
     {
-        require_once "Zend/Locale/Format.php";
+        return $this->cast($this->value);
+    }
 
-        if (!$this->isNumeric()) {
-            return 0;
+    /**
+     * Gets the current locale this field is set to.
+     *
+     * @return string
+     */
+    public function getLocale()
+    {
+        if ($this->locale) {
+            return $this->locale;
         }
 
-        $locale = new Zend_Locale($this->getLocale());
+        return i18n::get_locale();
+    }
 
-        $number = Zend_Locale_Format::getNumber(
-            $this->clean($this->value),
-            array('locale' => $locale)
-        );
+    /**
+     * Override the locale for this field.
+     *
+     * @param string $locale
+     *
+     * @return $this
+     */
+    public function setLocale($locale)
+    {
+        $this->locale = $locale;
 
-        return $number;
+        return $this;
     }
 
     /**
-     * Creates a read-only version of the field.
+     * Determine if we should use html5 number input
      *
-     * @return NumericField_Readonly
+     * @return bool
      */
-    public function performReadonlyTransformation()
```

Comment:
Somehow the readonly version doesn't respect `setLocale('de')` or `setScale(4)`. The disabled version is fine, weird. Example from a slightly modified frameworktest:

![image](https://cloud.githubusercontent.com/assets/111025/22910908/0ef0d4a2-f2c2-11e6-90d2-9dc7ad584e11.png)